spec/services/waste_exemptions_engine/confirmation_pdf_generator_service_spec.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+module WasteExemptionsEngine
+  RSpec.describe ConfirmationPdfGeneratorService do
+    describe ".send_confirmation_email" do
+      it "generates and return a string containing PDF content with the confirmation information" do
+        travel_to Time.local(2018, 1, 1, 1, 5, 0)
+        registration = create(:registration, :submitted, reference: "REFERENCE-1")
+        registration.addresses << create(:address, :site_address)
```

Comment:
We probably haven't spoken about it (and should probably write it down somewhere 😩 😳 ) but we agreed after the Waste Carriers work to adopt a pattern of all factory object creation being driven through traits, and to avoid doing any additional changes to the object in the tests.

So ideally adding the site address should be moved to the `:submitted` trait. And if 'submitted' no longer fits the bill because of this, possibly change the name (`:confirmable`, `:confirmation_pdf`, `:i_dont_know_cause_naming_things_sucks`).

Reply:
Nice! I'll address this :)